### PR TITLE
Leaves - Dominique

### PR DESCRIPTION
**Pull request by dtaylor73**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The Model communicates with the database and handles the "business logic."
Describe in your own words what the Controller is doing in Rails | The Controller receives messages from the router and manages the relationship between the view and model. It gathers data from the model and passes it to the View.
Describe in your own words what the View is doing in Rails | The View only handles "display logic" and renders data to HTML.
Describe an edge-case controller test you wrote | For the update method, I tested that the webpage would redirect to the homepage if given an invalid id.
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params provides a safe way to require and permit information that comes from forms. It is a convention that we should follow to DRY our code so we don't have to list each individual field of the new instance.
How are Rails migrations related to Rails models? | Migrations change the database schema.  Models communicate with a table in the schema. When you make a migration, you change how the model communicates with that tables.
Describe one area of Rails that are still unclear on | Testing syntax and the difference between testing the database, the show method and the index method. Also, I'm unclear on how to style html elements that are in the same line as ruby code.

### CODE REVIEW

File: app/views/tasks/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<p> <%= @task.description %> and should be completed
```
Comment (dHelmgren) [inline app/views/tasks/show.html.erb:1]: this is very much small potatoes but the phrase 'and should be complete' doesn't make a ton of sense here.

also, it would be nice to see the title of the task. :)

**Comment by dHelmgren** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes
Answered comprehension questions | yes
Successfully handles: Index, Show | yes
Index & Show tests pass | yes
Successfully handles: New, Create | yes
New & Create tests pass | yes
Successfully handles: Edit, Update |yes
Tests for Edit & Update test valid & invalid task ids | yes
Successfully handles: Destroy, Task Complete | yes
Tests for Destroy & Task Complete include tests for valid and invalid task ids | yes
Routes follow RESTful conventions | yes
Uses named routes (like `_path`) | yes
**Overall** | Good job on Task List! It functions as expected! You hit all of the learning goals: using partials, using strong params, and following Rails conventions.
